### ISSUE TITLE
Address form problem

### ISSUE THREAD

**Issue by Crumplezone** [tomatocart/TomatoCart-v1#273]
Some customers at [malbrad.co.uk/cart](url) have reported problems with the address form on checkout.
For example

can't get past the point, where I filled my address in. First I got a message saying, and I quote, " Choose State from drop down menu". It says United Kingdom, on the "Country" drop down menu. There is no "State" drop down menu?? 
Site is for UK only. It works fine for me, so I can't replicate the problem.

**Comment by martin03** [comment]
Hi. I checked the form of address when completing your order. the problem is duplicate records in cities.
For example:  Cumbria - 2, Devon - 2, Derbyshine - 2, etc.
Log into the database 'toc_zones' and check for duplicate records. Delete duplicate entries and re-enter through the admin menu of the store. This is a common problem, especially when you have a city and region. For example, Sofia and Sofia region. Duplication of characters does not allow to pass the address form.
Good luck

**Comment by Crumplezone** [comment]
Many thanks. I will do what you suggested. The problem was that many of the UK counties were incorrect, so I had to input them all manually.

**Comment by martin03** [comment]
Yes. It is best to be entered manually in order to avoid duplication of new
areas. Then check through a test account from the dropdown menu or directly
in the database if all is well and there is no duplication.
I'll be happy if it works for you.

2016-02-29 16:44 GMT+02:00 Crumplezone notifications@github.com:

**Comment by Crumplezone** [comment]
Where can you see the zone IDs -2 you mention? I have gone through the database and deleted duplicate area names, but this number is not in the database as below.

zone_id     zone_country_id     zone_code   zone_name
    Edit Edit   Copy Copy   Delete Delete   3790    222     ABD     Aberdeenshire
    Edit Edit   Copy Copy   Delete Delete   3791    222     ANG     Angus
    Edit Edit   Copy Copy   Delete Delete   3792    222     AVO     Avon
    Edit Edit   Copy Copy   Delete Delete   3794    222     ANT     Antrim
    Edit Edit   Copy Copy   Delete Delete   3795    222     ARG     Argyll
    Edit Edit   Copy Copy   Delete Delete   3796    222     ARM     Armagh
    Edit Edit   Copy Copy   Delete Delete   3797    222     AYR     Ayrshire

**Comment by martin03** [comment]
Are you sure that phpmyadmin - toc_zone all duplicate records are deleted?
Make good because when trying to choose a city, there are duplicate records.
When trying to choose any duplicate city name - returns an error.
But cities are not duplicated - the order continues to the next stage.

2016-03-01 19:38 GMT+02:00 Crumplezone notifications@github.com:

**Comment by martin03** [comment]
So. There is another problem. With the validation of email addresses.
Of introduced two real address displayed error. This will test it on my
shop to see where it could be .....

2016-03-02 16:00 GMT+02:00 Мартин Жеков martin03staff@gmail.com:

**Comment by Crumplezone** [comment]
Thanks for your help. I have definitely deleted all the duplicate zones now.

**Comment by martin03** [comment]
I hope this will help. It is better for business.
На 02.03.2016 20:55 "Crumplezone" notifications@github.com написа:
